app/controllers/director/counselors_controller.rb

```diff
@@ -29,6 +31,9 @@ def create
       )
       if @counselor.save
         puts "COUNSELOR SAVED"
+        @counselor.groups << @group
```

Comment:
Bring line 22 here. That way you don't do a DB search if the user and counselor don't save.

And what if the group doesn't exist? There is no error handling here.